src/types/redis_stream.cc

```diff
@@ -339,20 +339,36 @@ rocksdb::Status Stream::DeletePelEntries(const Slice &stream_name, const std::st
   WriteBatchLogData log_data(kRedisStream);
   batch->PutLogData(log_data.Encode());
 
+  std::map<std::string, uint64_t> consumer_acknowledges;
   for (const auto &id : entry_ids) {
     std::string entry_key = internalPelKeyFromGroupAndEntryId(ns_key, metadata, group_name, id);
     std::string value;
     s = storage_->Get(rocksdb::ReadOptions(), stream_cf_handle_, entry_key, &value);
     if (s.ok()) {
       *acknowledged += 1;
       batch->Delete(stream_cf_handle_, entry_key);
+
+      // increment ack for each related consumer
+      auto pel_entry = decodeStreamPelEntryValue(value);
+      consumer_acknowledges[pel_entry.consumer_name]++;
     }
   }
   if (*acknowledged > 0) {
     StreamConsumerGroupMetadata group_metadata = decodeStreamConsumerGroupMetadataValue(get_group_value);
     group_metadata.pending_number -= *acknowledged;
     std::string group_value = encodeStreamConsumerGroupMetadataValue(group_metadata);
     batch->Put(stream_cf_handle_, group_key, group_value);
+
+    for (const auto &consumer : consumer_acknowledges) {
+      auto consumer_meta_key = internalKeyFromConsumerName(ns_key, metadata, group_name, consumer.first);
+      std::string consumer_meta_original;
+      s = storage_->Get(rocksdb::ReadOptions(), stream_cf_handle_, consumer_meta_key, &consumer_meta_original);
+      if (s.ok()) {
```

Comment:
what if `!s.ok() && !s.IsNotFound()`?

Reply:
Thanks very for your correction.

It should be an internal error and I think that I need to add the error return value with `Status{Status::RedisExecErr, s.ToString()}`.

Do I need to add golang integration test for this error handling case? This may be hard to produce a case for it.

Reply:
> Do I need to add golang integration test for this error handling case? 

Currently it's a bit hard to modify this, especially in integration tests. We can just handle this without testing.

Reply:
Got it! I have corrected my wording above.

This function returns `rocksdb::Status` and I think I should return the status immediately in the function when `!s.ok() && !s.IsNotFound()`.

Reply:
There're two gets here, right? `if (s.ok())` in the loop should also adds this checks(Though errors merely happens)

Reply:
Thanks for your correction!

Sorry for missing another error handling. I have added it with a new commit.